src/utils/file-processor.js

```diff
@@ -175,31 +88,34 @@ const fileToChunks = (file, handle) =>
   });
 
 // Pipeline: chunks |> fromTrytes |> decrypt |> combineChunks
-const chunksToFile = (chunks, handle) =>
+const chunksToFile = (chunks, handle, opts = {}) =>
   new Promise((resolve, reject) => {
     try {
+      // Remove metachunk
+      if (opts.withMeta) chunks.splice(0, 1);
+
+      // ASC order.
+      // NOTE: Cannot use `>` because JS treats 0 as null and doesn't work.
+      chunks.sort((x, y) => x.idx - y.idx);
+
       const bytes = chunks
+        .map(({ data }) => data)
         .map(Iota.utils.fromTrytes)
         .map(chunk => Encryption.decrypt(chunk, handle)) // treasure => null
         .join(""); // join removes nulls
 
       resolve(new Blob([bytes]));
     } catch (err) {
+      console.log("IN HEEERRRR");
```

Comment:
This is the only thing I noticed.  Approving anyway since I'm making a webinterface PR shortly and I'll just pull this out in mine.